retrofit/src/main/java/retrofit2/RequestBuilder.java

```diff
@@ -138,6 +134,17 @@ private static void canonicalize(Buffer out, String input, int pos, int limit,
     }
   }
 
+  private static boolean shouldSkipCodePoint(final int codePoint) {
```

Comment:
Interesting. I guess i'm surprised that _runtime_ replacement of whitespace characters helps paper transcription. My thoughts was that it was to prevent request splitting type issues. (not xss)
Do you have a better name in mind? perhaps `shouldSkipHardToTranscribeCodePoints`? i can add a comment capturing your thoughts above.